### DISCUSSION TITLE
Does CVE-2024-6387 affect opennssh server on windows?

### DISCUSSION THREAD

**Discussion by shimizu-n** [PowerShell/Win32-OpenSSH#2248] [Q&A] [answered]
Does CVE-2024-6387 affect opennssh server on windows?

**Comment by MarcelNullVier** [comment]
Qualys currently states OpenSSH versions starting 8.5p1 and prior to 9.8p1 are vulnerable. For Windows: Detection relies on the HKLM\SYSTEM\CurrentControlSet\services\sshd registry key to get the sshd.exe file path and checks for file version information from sshd.exe file.

We are also curious, if Windows is affected.

**Comment by NoMoreFood** [comment]
On the surface, it doesn't appear like the Windows builds _should_ be affected given that the signal handlers were rewritten for Windows and don't call the function stated in the various disclosures.  It would be good for someone with my recent time on the project from Microsoft to confirm my assertion though.  Regardless, I suspect Microsoft may just patch it anyhow to avoid having the argument.

**Comment by rgoltz** [comment]
Issue here in this repo: https://github.com/PowerShell/Win32-OpenSSH/issues/2249

**Comment by RivasMario** [comment]
How does one upgrade to 9.8. I can't find any build for win past 9.5 atm. Ty

**Comment by FrancoisSSC** [comment, marked as answer]
I sent an email to secure@microsoft.com earlier today, and this is their official statement:

https://github.com/PowerShell/Announcements/issues/63
https://msrc.microsoft.com/update-guide/vulnerability/CVE-2024-6387

You can all sleep well now! :)
